extensions/wikia/AdEngine/js/video/player/porvata/floater.js

```diff
@@ -0,0 +1,174 @@
+		'wikia.window',
+		'wikia.document'
+	], function (win, doc) {
+		'use strict';
+
+		var activeFloatingCssClass = 'floating',
+			compatibleSlots = ['TOP_LEADERBOARD'],
+			floatingVideoPadding = 28;
+
+		function resetAdContainer(top) {
+			top.style.removeProperty('width');
+			top.style.removeProperty('height');
+		}
+
+		function updateDimensions(element, width, height) {
+			if (element) {
+				element.style.width = width + 'px';
+				element.style.height = height + 'px';
+			}
+		}
+
+		function updateImage(image, width, height) {
+			if (image) {
+				image.width = width;
+				image.style.marginTop = ((height - image.height) / 2) + 'px';
+			}
+		}
+
+		function createCloseButton() {
+			var a = doc.createElement('a');
+
+			a.classList.add('floating-close-button');
+			a.classList.add('hidden');
+
+			return a;
+		}
+
+		function createOnCloseListener(elements, params, floatingContext, imageMarginTop) {
+			return function () {
+				disableFloating(elements, params, imageMarginTop);
+				win.removeEventListener('scroll', floatingContext.scrollListener);
+
+				if (floatingContext.onClose) {
+					floatingContext.onClose();
+				}
+
+				deleteCloseButton(elements, floatingContext);
+			};
+		}
+
+		function deleteCloseButton(elements, floatingContext) {
+			if (floatingContext.closeButton) {
+				elements.top.removeChild(floatingContext.closeButton);
+			}
+		}
+
+		function resetDimensions(element, params) {
+			updateDimensions(element, params.width, params.height);
+		}
+
+		function resetImage(image, marginTop) {
+			if (image) {
+				image.removeAttribute('width');
+				image.style.marginTop = marginTop;
+			}
+		}
+
+		function enableFloating(elements) {
+			var width = ((win.innerWidth - elements.background.offsetWidth) / 2) - floatingVideoPadding,
+				height = width;
+
+			elements.top.classList.toggle(activeFloatingCssClass);
+			updateDimensions(elements.top, width, height);
+			updateDimensions(elements.iframe, width, height);
+			updateDimensions(elements.imageContainer, width, height);
+			updateImage(elements.image, width, height);
+			elements.video.resize(width, height);
+		}
+
+		function disableFloating(elements, params, imageMarginTop) {
+			elements.top.classList.toggle(activeFloatingCssClass);
+			resetAdContainer(elements.top);
+			resetDimensions(elements.iframe, params);
+			resetDimensions(elements.imageContainer, params);
+			resetImage(elements.image, imageMarginTop);
+			elements.video.resize(params.width, params.height);
+		}
+
+		function enableFloatingOn(video, params, onClose) {
+			var top = doc.getElementById('WikiaTopAds'),
+				threshold = 100,
+				scrollYOffset = top.offsetTop + top.offsetHeight + threshold,
+				imageMarginTop,
+				floatingContext = {
+					onClose: onClose
+				},
+				imageContainer = params.container.parentElement.querySelector('#image'),
+				elements = {
+					top: top,
+					background: doc.getElementById('WikiaPageBackground'),
+					iframe: top.querySelector('iframe'),
```

Comment:
Is it safe? You're loooking for TLB iframe, am I right? What about taking the iframe from params.container element?